src/smriprep/cli/run.py

```diff
@@ -495,6 +528,32 @@ def build_workflow(opts, retval):
     bids_dir = opts.bids_dir.resolve()
     layout = BIDSLayout(str(bids_dir), validate=False)
     subject_list = collect_participants(layout, participant_label=opts.participant_label)
+    session_list = opts.session_id or []
+
+    subject_session_list = []
+    for subject in subject_list:
+        sessions = layout.get_sessions(
+            scope='raw',
+            subject=subject,
+            session=session_list or Query.OPTIONAL,
+            suffix=['T1w', 'T2w'],  # TODO: Track supported modalities globally
+        )
+        if not sessions:
+            if opts.subject_anatomical_reference == 'sessionwise':
+                logger.warning(
+                    '--subject-anatomical-reference "sessionwise" was requested, but no sessions '
+                    f'were found for subject {subject}.'
+                )
+            subject_session_list.append((subject, None))
+            continue
```

Comment:
I'd rather raise if sessionwise doesn't apply. Also, it should also apply is there's only one session, right?
```suggestion
        if not sessions or len(sessions) == 1 and opts.subject_anatomical_reference == 'sessionwise':
            raise RuntimeError(
                '--subject-anatomical-reference "sessionwise" was requested, but no sessions '
                f'were found for subject {subject}.'
            )
```

Reply:
I think we want to fail in this case to be explicit, the behavior in case of a single-session is clear, and this opens it up to having users who ignore the warning and even do not figure out that their dataset is not as they are assuming it is (e.g., because they are just copying the fmriprep command line from a previous project).

IMHO we should keep our stance of not being lenient in these situations. If you want to add a explicit bypass to this exception, I'd be fine (like the bids validation at the beginning -- we fail, but user can deliberately bypass it).

Reply:
Also, my suggestion simplifies the logic, removing one nesting level.

Reply:
i'm fine with starting with an error, and then re-evaluating if it becomes burdensome